third_party/bigframes_vendored/pandas/core/computation/ops.py

```diff
@@ -0,0 +1,605 @@
+# Contains code from https://github.com/pandas-dev/pandas/blob/main/pandas/core/computation/ops.py
+"""
+Operator classes for eval.
+"""
+
+from __future__ import annotations
+
+from datetime import datetime
+from functools import partial
+import operator
+from typing import Callable, Literal, TYPE_CHECKING
+
+import bigframes_vendored.pandas.core.common as com
+from bigframes_vendored.pandas.core.computation.common import (
+    ensure_decoded,
+    result_type_many,
+)
+from bigframes_vendored.pandas.core.computation.scope import DEFAULT_GLOBALS
+import numpy as np
+from pandas._libs.tslibs import Timestamp
+from pandas.core.dtypes.common import is_list_like, is_scalar
+from pandas.io.formats.printing import pprint_thing, pprint_thing_encoded
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Iterator
+
+REDUCTIONS = ("sum", "prod", "min", "max")
+
+_unary_math_ops = (
+    "sin",
+    "cos",
+    "exp",
+    "log",
+    "expm1",
+    "log1p",
+    "sqrt",
+    "sinh",
+    "cosh",
+    "tanh",
+    "arcsin",
+    "arccos",
+    "arctan",
+    "arccosh",
+    "arcsinh",
+    "arctanh",
+    "abs",
+    "log10",
+    "floor",
+    "ceil",
+)
+_binary_math_ops = ("arctan2",)
```

Comment:
Aside: are there any of these unary math ops that we're missing in our series implementation?

Reply:
We are missing `expm1`, `log1p`, `floor`, `ceil` and `arctan2`. Will see what I can do about adding those asap.